tests/test_smbclient_shutil.py

```diff
@@ -71,6 +74,14 @@ def _set_file_attributes(path, attributes):
             set_info(transaction, basic_info)
 
 
+def test_basename_local():
```

Comment:
It might be better to test one of the functions you are hoping would accept a local path instead of just testing this function directly. Same for join below.

Reply:
I...guess? I sort of thought `copyfile()` already did accept local paths. The docstring says it does. I didn't think this particular PR actually changed any functionality. The point was just to extract the conditional from `_copy` to a reusable convenience function.

Reply:
This sort of ultimately leads into my comment here https://github.com/jborean93/smbprotocol/pull/106#issuecomment-834935960. There's a few PRs that are creating/moving helper functions with the ultimate goal of supporting remote <-> local copytree functionality. You are better off trying to do that one goal in 1 PR unless the amount of changes is quite drastic. It helps you to have 1 branch that tracks all your changes and to actually test these helper functions as part of what they are inteded for.